hudi-client/hudi-client-common/src/main/java/org/apache/hudi/client/BaseHoodieWriteClient.java

```diff
@@ -588,6 +588,19 @@ protected void runTableServicesInline(HoodieTable table, HoodieCommitMetadata me
         metadata.addMetadata(HoodieClusteringConfig.SCHEDULE_INLINE_CLUSTERING.key(), "true");
         inlineScheduleClustering(extraMetadata);
       }
+
+      // if clustering is disabled, but we might need to rollback any inflight clustering when clustering was enabled previously.
+      if (!config.inlineClusteringEnabled() && !config.isAsyncClusteringEnabled() && !config.scheduleInlineClustering()
```

Comment:
Could this condition simplify to `config.isRollbackPendingClustering()`? IMO, regardless of whether Clustering is enabled or not, only setting hoodie.rollback.pending.clustering to true, it should rollback any pending clustering, meanwhile, if disable clustering previously, there is no any pending clustering.

Reply:
nope. if clustering is enabled, we don't want to rollback pending clustering. could be async clustering running. thats why.